framework/DataObjects/TestHierarchal.py

```diff
@@ -0,0 +1,409 @@
+import warnings
+warnings.simplefilter('default',DeprecationWarning)
+
+import xml.etree.ElementTree as ET
+import sys, os
+import pickle as pk
+import numpy as np
+import xarray as xr
+frameworkDir = os.path.dirname(os.path.abspath(os.path.join(sys.argv[0],'..')))
+
+sys.path.append(frameworkDir)
+from utils.utils import find_crow
+
+find_crow(frameworkDir)
+
+import XDataSet
+import MessageHandler
+
+mh = MessageHandler.MessageHandler()
+mh.initialize({'verbosity':'silent', 'callerLength':10, 'tagLength':10})
+
+def createElement(tag,attrib=None,text=None):
+  """
+    Method to create a dummy xml element readable by the distribution classes
+    @ In, tag, string, the node tag
+    @ In, attrib, dict, optional, the attribute of the xml node
+    @ In, text, str, optional, the dict containig what should be in the xml text
+  """
+  if attrib is None:
+    attrib = {}
+  if text is None:
+    text = ''
+  element = ET.Element(tag,attrib)
+  element.text = text
+  return element
+
+results = {"pass":0,"fail":0}
+
+def checkFloat(comment,value,expected,tol=1e-10,update=True):
+  """
+    This method is aimed to compare two floats given a certain tolerance
+    @ In, comment, string, a comment printed out if it fails
+    @ In, value, float, the value to compare
+    @ In, expected, float, the expected value
+    @ In, tol, float, optional, the tolerance
+    @ In, update, bool, optional, if False then don't update results counter
+    @ Out, res, bool, True if same
+  """
+  if np.isnan(value) and np.isnan(expected):
+    res = True
+  elif np.isnan(value) or np.isnan(expected):
+    res = False
+  else:
+    res = abs(value - expected) <= tol
+  if update:
+    if not res:
+      print("checking float",comment,'|',value,"!=",expected)
+      results["fail"] += 1
+    else:
+      results["pass"] += 1
+  return res
+
+def checkTrue(comment,res,update=True):
+  """
+    This method is a pass-through for consistency and updating
+    @ In, comment, string, a comment printed out if it fails
+    @ In, res, bool, the tested value
+    @ In, update, bool, optional, if False then don't update results counter
+    @ Out, res, bool, True if test
+  """
+  if update:
+    if res:
+      results["pass"] += 1
+    else:
+      print("checking bool",comment,'|',res,'is not True!')
+      results["fail"] += 1
+  return res
+
+def checkSame(comment,value,expected,update=True):
+  """
+    This method is aimed to compare two identical things
+    @ In, comment, string, a comment printed out if it fails
+    @ In, value, float, the value to compare
+    @ In, expected, float, the expected value
+    @ In, update, bool, optional, if False then don't update results counter
+    @ Out, res, bool, True if same
+  """
+  res = value == expected
+  if update:
+    if res:
+      results["pass"] += 1
+    else:
+      print("checking string",comment,'|',value,"!=",expected)
+      results["fail"] += 1
+  return res
```

Comment:
probably since we are repeating these methods in each unit test, we should collect them in a single module and import them.

Reply:
I agree but it will take passing in "results" and changing all the method calls. It should be done, but it's a bit of a pain.